### PR TITLE
BUG: pd.DataFrame.transform recursively loops in some cases #34224

### LINKED ISSUE #34224: BUG: pd.DataFrame.transform recursively loops in some cases
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
pd.DataFrame({"a":[None]}).transform({"a":int})
```

#### Problem description

Executing the above causes recursion depth limit exception. This is confusing and it is harder to pinpoint/debug than the expected exception.

#### Expected Output

Something akin to the output of `int(None)`

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.6.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.0.3
numpy            : 1.18.4
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 19.2.3
setuptools       : 41.2.0
Cython           : None
pytest           : 3.10.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 3.10.1
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by pedrooa**
- [x] closes #34224 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Error handling within try/except in aggregate function of frame.py file

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
can you add the test which hits this

**Comment by pedrooa** [comment]
> can you add the test which hits this

Sure! Do You mean singling out a test that already exists and hits this, or creating a new test for this case?

**Comment by dsaxton** [comment]
> Sure! Do You mean singling out a test that already exists and hits this, or creating a new test for this case?

You'll want to create a new test for a case that was raising the RecursionError before, but now raises a more appropriate error / message. You could use the example from the original issue.

File: pandas/tests/series/test_apply.py [hunk 1]
```diff
@@ -459,6 +463,12 @@ def test_agg_cython_table_raises(self, series, func, expected):
             # e.g. Series('a b'.split()).cumprod() will raise
             series.agg(func)
 
+    def test_transform_none_to_type(self):
+        df = pd.DataFrame({"a": [None]})
+
+        with pytest.raises(TypeError):
+            df.transform({"a": int})
```
Comment (dsaxton) [inline pandas/tests/series/test_apply.py:470]: Can you also include the issue number as a comment and test the new error message?

**Review by jreback — CHANGES_REQUESTED** [review]
looks good can u add a whats new note
reshaping bug fix for 1.1

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green.

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -960,6 +960,7 @@ Reshaping
 - Bug in :func:`concat` was not allowing for concatenation of ``DataFrame`` and ``Series`` with duplicate keys (:issue:`33654`)
 - Bug in :func:`cut` raised an error when non-unique labels (:issue:`33141`)
 - Ensure only named functions can be used in :func:`eval()` (:issue:`32460`)
+- Bug in :func:`aggregate` was causing recursive loop in some cases (:issue:`34224`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:963]: actually this is not right, can you put `` :func:`DataFrame.aggregate` `` and `` :func:`Series.aggregate` ``.

pls ping on green.

**Comment by jreback** [comment]
thanks @pedrooa

**Pull request merged**